### DISCUSSION TITLE
operator |= not defined

### DISCUSSION THREAD

**Discussion by sudokit** [vlang/v#21966] [Questions and Answers]
Didnt want to open an issue, since nto sure if its and issue, but should this be defined?

Something like this should work, right?:
```v
module main

fn main() {
	mut foo := false
	bar := true

	foo |= bar
}
```
but it errors with:
```
t.v:7:2: error: operator |= not defined on left operand type `bool`
    5 |     bar := true
    6 |
    7 |     foo |= bar
      |     ~~~
    8 | }
```

**Comment by JalonSolov** [comment]
As documented here:  https://docs.vlang.io/appendix-ii-operators.html

`|` is bitwise OR that works on integers.

`||` is logical OR that works on bools.
```v
module main

fn main() {
	mut foo := false
	bar := true

	foo ||= bar

	println(foo)
}
```
works as expected.

**Reply by sudokit** [reply]
Aaaah right my bad i just saw |= and though its for all. Thanks for the clarification!
